x-pack/plugins/security_solution/public/management/pages/policy/view/policy_settings_form/components/detect_prevent_protection_lavel.tsx

```diff
@@ -0,0 +1,188 @@
+          label: PREVENT_LABEL,
+          flexGrow: 5,
+        },
+      ];
+    }, []);
+
+    const currentProtectionLevelLabel = useMemo(() => {
+      const radio = radios.find((item) => item.id === policy.windows[protection].mode);
+
+      if (radio) {
+        return radio.label;
+      }
+
+      return PREVENT_LABEL;
+    }, [policy.windows, protection, radios]);
+
+    return (
+      <div data-test-subj={getTestId()}>
+        <SettingCardHeader>
+          <FormattedMessage
+            id="xpack.securitySolution.endpoint.policyDetailsConfig.protectionLevel"
+            defaultMessage="Protection level"
+          />
+        </SettingCardHeader>
+        <EuiSpacer size="xs" />
+        <EuiFlexGroup>
+          {isEditMode ? (
+            radios.map(({ label, id, flexGrow }) => {
+              return (
+                <EuiFlexItem grow={flexGrow} key={id}>
+                  <ProtectionRadio
+                    policy={policy}
+                    onChange={onChange}
+                    mode={mode}
+                    protection={protection}
+                    protectionMode={id}
+                    osList={osList}
+                    label={label}
+                    data-test-subj={getTestId(`${id}Radio`)}
+                  />
+                </EuiFlexItem>
+              );
+            })
+          ) : (
+            <>{currentProtectionLevelLabel}</>
+          )}
+        </EuiFlexGroup>
+      </div>
+    );
+  }
+);
+DetectPreventProtectionLevel.displayName = 'DetectPreventProtectionLevel';
+
+interface ProtectionRadioProps extends PolicyFormComponentCommonProps {
+  protection: PolicyProtection;
+  protectionMode: ProtectionModes;
+  osList: ImmutableArray<Partial<keyof UIPolicyConfig>>;
+  label: string;
+}
+
+const ProtectionRadio = React.memo(
+  ({
+    protection,
+    protectionMode,
+    osList,
+    label,
+    onChange,
+    policy,
+    mode,
+    'data-test-subj': dataTestSubj,
+  }: ProtectionRadioProps) => {
+    const radioButtonId = useGeneratedHtmlId();
+    const selected = policy.windows[protection].mode;
+    const isPlatinumPlus = useLicense().isPlatinumPlus();
+    const showEditableFormFields = mode === 'edit';
+
+    const handleRadioChange = useCallback(() => {
+      const newPayload = cloneDeep(policy);
+
+      for (const os of osList) {
+        if (os === 'windows') {
+          newPayload[os][protection].mode = protectionMode;
+        } else if (os === 'mac') {
+          newPayload[os][protection as MacPolicyProtection].mode = protectionMode;
+        } else if (os === 'linux') {
+          newPayload[os][protection as LinuxPolicyProtection].mode = protectionMode;
+        }
+        if (isPlatinumPlus) {
+          if (os === 'windows') {
+            if (protectionMode === ProtectionModes.prevent) {
+              newPayload[os].popup[protection].enabled = true;
+            } else {
+              newPayload[os].popup[protection].enabled = false;
+            }
```

Comment:
nit: This can also be written as 
```json5
newPayload[os].popup[protection].enabled = protectionMode === ProtectionModes.prevent;
```

Same goes for the other ones below if you change it.